.github/workflows/backend-cd.yml

```
@@ -0,0 +1,56 @@
+name: Backend CD Workflow
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ "main", "dev" ]
+
+jobs:
+  Build:
+    name: 🏗️ Build Jar and Upload Docker Image
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'corretto'
+
+      - name: 🏗️ Set up Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: 🏗️ Build with Gradle
+        run: ./gradlew clean bootJar
+
+      - name: 🐳 Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      - name: 🐳 Docker Image Build and Push
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          push: true
+          tags: ${{ secrets.DOCKERHUB_USERNAME }}/${{ secrets.DOCKER_IMAGE_NAME_DEV }}:${{ github.sha }}
+          platforms: linux/amd64
+
+  Deploy:
+    needs: Build
+    name: 🚀 Deployment
+    runs-on: [ self-hosted, dev ]
+
+    steps:
+      - name: 🚀 Deployment
+        run: |
+          docker login -u ${{ secrets.DOCKERHUB_USERNAME }} -p ${{ secrets.DOCKERHUB_PASSWORD }}
+          docker stop ${{ secrets.DOCKER_CONTAINER_NAME }} | true
+          docker rm ${{ secrets.DOCKER_CONTAINER_NAME }} | true
+          docker image prune -a -f
+          docker pull ${{ secrets.DOCKERHUB_USERNAME }}/${{ secrets.DOCKER_IMAGE_NAME_DEV }}:${{ github.sha }}
+          docker run --name ${{ secrets.DOCKER_CONTAINER_NAME }} -d -v /logs:/logs ${{ secrets.DOCKERHUB_USERNAME }}/${{ secrets.DOCKER_IMAGE_NAME_DEV }}:${{ github.sha }}
```

```suggestion
      - name: Login to Docker Hub
        uses: docker/login-action@v3
        with:
          username: ${{ secrets.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}

      - name: Docker Compose up
        run: docker compose -f compose.dev.yml up -d

      - name: Clean Unused Image
        run: docker image prune -af
```

이런 방식으로 사용하는 방법도 있습니다!